finagle-thrift/src/main/ruby/lib/finagle-thrift/trace.rb

```diff
@@ -57,6 +57,10 @@ def tracer=(tracer)
     @tracer = tracer
   end
 
+  def stack=(stack)
```

Comment:
@mosesn indeed it is a thread local variable.  I hope I'm understanding your question correctly, I think we are trying to do one trace object per thread.  We were getting race conditions on the stack obj.  I think Ruby does context switching from what I read [here](https://ruby-hacking-guide.github.io/thread.html), but since its thread local it shouldn't be problem.  Also, we're using jruby so they should be real threads.